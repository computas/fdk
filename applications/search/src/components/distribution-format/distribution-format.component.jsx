import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

export const DistributionFormat = props => {
  const { code, type, text } = props;
  const formatClass = cx('fdk-label-distribution mr-2', {
    'fdk-label-distribution-offentlig': code === 'PUBLIC',
    'fdk-label-distribution-begrenset': code === 'RESTRICTED',
    'fdk-label-distribution-skjermet': code === 'NON-PUBLIC'
  });

  // <i className="fa fa-cogs fdk-fa-left" /> TYPE ICON

  return (
    <div className={formatClass}>
      {type && (
        <span>
          <strong className="fdk-distribution-format">{type}</strong>
        </span>
      )}
      {text}
    </div>
  );
};

DistributionFormat.defaultProps = {
  code: 'PUBLIC',
  type: null,
  text: null
};

DistributionFormat.propTypes = {
  code: PropTypes.string,
  type: PropTypes.string,
  text: PropTypes.string
};
