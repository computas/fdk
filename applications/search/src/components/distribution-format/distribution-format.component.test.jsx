import React from 'react';
import { shallow } from 'enzyme';
import { DistributionFormat } from './distribution-format.component';

test('should render DistributionFormat with "Dataset" correctly', () => {
  const defaultProps = {
    code: 'PUBLIC',
    text: 'ESRI Filgeodatabase',
    type: 'xml'
  };
  const wrapper = shallow(<DistributionFormat {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render DistributionFormat with "Dataset" correctly with no type', () => {
  const defaultProps = {
    code: 'PUBLIC',
    text: 'ESRI Filgeodatabase'
  };
  const wrapper = shallow(<DistributionFormat {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});
