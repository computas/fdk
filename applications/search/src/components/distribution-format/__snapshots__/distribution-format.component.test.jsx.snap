// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DistributionFormat with "Dataset" correctly 1`] = `
<div
  className="fdk-label-distribution mr-2 fdk-label-distribution-offentlig"
>
  <span>
    <strong
      className="fdk-distribution-format"
    >
      xml
    </strong>
  </span>
  ESRI Filgeodatabase
</div>
`;

exports[`should render DistributionFormat with "Dataset" correctly with no type 1`] = `
<div
  className="fdk-label-distribution mr-2 fdk-label-distribution-offentlig"
>
  ESRI Filgeodatabase
</div>
`;
